packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -9,15 +9,37 @@ export default function(api, opts) {
 
   function hasRestElement(path) {
     let foundRestElement = false;
-    path.traverse({
-      RestElement() {
-        foundRestElement = true;
-        path.stop();
-      },
+    visitRestElements(path, () => {
+      foundRestElement = true;
+      return /* stop */ true;
     });
     return foundRestElement;
   }
 
+  function visitRestElements(path, visitor) {
```

Comment:
I realize that this is probably more performant solution, but the code itself is imho a little bit convoluted, hard to read and to reason about. It has a lot of indirection in it. Would it be possible to rephrase it into a simpler implementation?